Definition spoofing support.

david-batranu opened the pull request:
Also works for direct entries as the word is part of the URL and can be used to fetch a spoofed definition (line `187`).

The first `<b>` tag of the definition `htmlRep` is replaced by the word that was searched for. All custom markup is lost as a result. (`<sup>`, accents...).

Setting `spoofNormalize` to `false` will give more control over the definition, for each form of the word, but will also require defining a mapping for each form.

CatalinFrancu commented on wwwbase/search.php:
```diff
@@ -87,6 +94,14 @@
 
+if(SPOOF_ENABLED && $cuv) {
+  $cuv_normalized = SPOOF_NORMALIZE ? mb_strtolower(StringUtil::unicodeToLatin($cuv)) : $cuv;
+  $cuv_spoofed = SPOOF_WORDS[$cuv_normalized];
```
Please add a `@` here (as in `@SPOOF_WORDS[...`). This suppresses a warning in case no spoof words are defined (i.e., the `spoofWords` section is commented out in `dex.conf`).

CatalinFrancu commented on wwwbase/search.php:
```diff
@@ -236,6 +263,21 @@
+if(SPOOF_ENABLED && $cuv_spoofed) {
+
+  function replaceSpoofedWord($definition, $cuv) {
+    $pattern = '/<b>(.*)<\/b>(\s.*)/';
```
Please add a `/U` modified (ungreedy), other wise the regexp will match everything to the last `</b>`).

CatalinFrancu commented on wwwbase/search.php:
```diff
@@ -236,6 +263,21 @@
+  function replaceSpoofedWord($definition, $cuv) {
+    $pattern = '/<b>(.*)<\/b>(\s.*)/';
+    $replacement = sprintf('<b>%s</b>${2}', mb_strtoupper($cuv));
+    $definition->htmlRep = preg_replace($pattern, $replacement, $definition->htmlRep);
```
...and a `, 1` last argument here to ensure that only the first `<b>` tag gets replaced.

CatalinFrancu commented:
Great! I played with it and spotted a couple of bugs, please see my comments. Once fixed, you should be able to merge the PR yourself (I think).

david-batranu commented:
Thanks for feedback. Implemented the requested changes. Also added `?` to the `*` in the second group of the regex as that has to be greedy now that `/U` is used.
